src/Umbraco.Web.Common/Macros/PartialViewMacroViewComponent.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Umbraco.Web.Models;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Umbraco.Core.Composing;
+using Umbraco.Core.Models.PublishedContent;
+using Umbraco.Web.Mvc;
+
+namespace Umbraco.Web.Macros
+{
+    /// <summary>
+    /// Controller to render macro content for Partial View Macros
+    /// </summary>
+    //[MergeParentContextViewData] // TODO is this requeired now it is a ViewComponent?
```

Comment:
This depends on my prev question and whether view contexts are nested and wired up, i have no idea how view components now work. I'm hoping a lot has changed but who knows. This is getting into that 'mvc magic' part of our RFC and we're going to run into a lot of this now since we do a lot of old 'mvc magic'. This all has to do with wiring up validation and ModelState during post backs. That in itself with SurfaceControllers and front-end routes is going to be challenging. It took me quite some time to get that all working with MVC. I worked with MS on making that work with dnx (< netcore 1) and it was weird. Now with endpoint routing it will be all changed again. I really want to work on that part. We won't really know abuot this stuff until we dive into it